src/lock.cr

```diff
@@ -32,13 +33,18 @@ module Shards
             case key = pull.read_scalar
             when "shards"
               pull.each_in_mapping do
-                dependencies << Dependency.from_yaml(pull, is_lock: true)
+                dep = Dependency.from_yaml(pull)
```

Comment:
I think we should add a `Package.from_yaml`. The `Dependency#as_package` might call the resolver while trying to convert it to a package and that does not seems right to do here.

Reply:
The call to the resolver is done only when converting from an old `shard.lock` file. Otherwise it cannot be converted to a plain `Version`. In the future we could just deprecate lock files with version `1.0` and the `Package` could be parsed directly from the yaml.

Reply:
Ok, Let's add a comment there to keep track of that.

Reply:
Comments added